### ISSUE TITLE
Interact with VSCode instantly, without waiting for full workspace to load

### ISSUE THREAD

**Issue by shaal** [gitpod-io/gitpod#7441]
## Is your feature request related to a problem? Please describe

It can take over a minute for a workspace with a large custom image or heavy prebuild artifacts to load. During that time, all you can do is watch a Gitpod logo jump.

## Describe the behaviour you'd like

Gitpod should display VSCode right away, without waiting for the workspace to fully load, allowing the user to interact with VSCode, similar to Github's experience when pressing `.` and getting VSCode displayed right away.

A workspace loading status should be displayed in a dedicated status panel, and perhaps certain panels should be restricted (i.e. Terminal) until the full workspace has loaded.

**Comment by akosyakov** [comment]
@loujaybee it reminds me that we had idea of progressive loading for long time. Not something to focus right now, but ideas for future.

**Comment by axonasif** [comment]
It will be really awesome when this is a reality. Although this might create the demand of more hardware resources.

**Comment by sr229** [comment]
I think this is possible now given the recent developments with github.dev and vscode.dev.

I think what we can do here in DX is this:

- Offer a basic VSCode Web Editor right off the bat. It won't provided advanced intellisense and not all extensions work.

- If the user specifically requests for a full workspace, we provide a notification popup detailing the progress of the full workspace then let VSCode for Web switch context to the full Gitpod Workspace.

**Comment by stemount** [comment]
Playing devil's advocate, whilst I agree GitPod's loading screen is way too vague and doesn't help you debug a slow starting workspace (is it my machine, GitPod remote, has my workspace crashed etc)... are there actually any browser speed/load benefits of painting a more complete IDE?

I've only ever had a theory as its mostly loading my OS into RAM at this time more than anything but not checked

**Comment by akosyakov** [comment]
> Offer a basic VSCode Web Editor right off the bat. It won't provided advanced intellisense and not all extensions work.

Yeah, VS Code not the issue, but prerequisite that we should allow to fetch files from any workspace on backup, i.e. virtual file system for workspace backups. github.dev using GH api and vscode.dev using local system.

**Comment by loujaybee** [comment]
Related to: 

- https://github.com/gitpod-io/gitpod/issues/8971

**Comment by shaal** [comment]
Please add `meta: never-stale` label to this issue
